src/flamenco/runtime/fd_executor.c

```diff
@@ -384,23 +384,36 @@ fd_executor_setup_accessed_accounts_for_txn( fd_exec_txn_ctx_t * txn_ctx ) {
         fd_txn_acct_addr_lut_t const * addr_lut = &addr_luts[i];
         fd_pubkey_t const * addr_lut_acc = (fd_pubkey_t *)((uchar *)txn_ctx->_txn_raw->raw + addr_lut->addr_off);
 
+        /* https://github.com/firedancer-io/agave/blob/368ea563c423b0a85cc317891187e15c9a321521/accounts-db/src/accounts.rs#L90-L94 */
         FD_BORROWED_ACCOUNT_DECL(addr_lut_rec);
         int err = fd_acc_mgr_view(txn_ctx->slot_ctx->acc_mgr, txn_ctx->slot_ctx->funk_txn, (fd_pubkey_t *) addr_lut_acc, addr_lut_rec);
         if( FD_UNLIKELY( err != FD_ACC_MGR_SUCCESS ) ) {
-          FD_LOG_ERR(( "addr lut not found" )); // TODO: return txn err code
+          return FD_RUNTIME_TXN_ERR_ADDRESS_LOOKUP_TABLE_NOT_FOUND;
+        }
+
+        /* https://github.com/firedancer-io/agave/blob/368ea563c423b0a85cc317891187e15c9a321521/accounts-db/src/accounts.rs#L96
+           https://github.com/firedancer-io/agave/blob/368ea563c423b0a85cc317891187e15c9a321521/accounts-db/src/accounts.rs#L114 */
+        if( FD_UNLIKELY( memcmp( addr_lut_rec->const_meta->info.owner, fd_solana_address_lookup_table_program_id.key, sizeof(fd_pubkey_t) ) ) ) {
+          return FD_RUNTIME_TXN_ERR_INVALID_ADDRESS_LOOKUP_TABLE_OWNER;
         }
 
+        /* https://github.com/anza-xyz/agave/blob/574bae8fefc0ed256b55340b9d87b7689bcdf222/accounts-db/src/accounts.rs#L141-L142 */
         fd_address_lookup_table_state_t addr_lookup_table_state;
         fd_bincode_decode_ctx_t decode_ctx = {
           .data = addr_lut_rec->const_data,
-          .dataend = &addr_lut_rec->const_data[56], // TODO macro const.
+          .dataend = &addr_lut_rec->const_data[FD_LOOKUP_TABLE_META_SIZE],
           .valloc  = fd_scratch_virtual(),
         };
-        if( fd_address_lookup_table_state_decode( &addr_lookup_table_state, &decode_ctx ) ) {
-          FD_LOG_ERR(("fd_address_lookup_table_state_decode failed"));
+
+        /* https://github.com/firedancer-io/agave/blob/368ea563c423b0a85cc317891187e15c9a321521/sdk/program/src/address_lookup_table/state.rs#L197-L198
+           https://github.com/firedancer-io/agave/blob/368ea563c423b0a85cc317891187e15c9a321521/sdk/program/src/address_lookup_table/state.rs#L205-L214 */
```

Comment:
why are there 2 comments here with differnet lines

Reply:
Because the logic is split up in a weird if condition